samples/contrib/aws-samples/README.md

```diff
@@ -0,0 +1,197 @@
+      [Example](https://github.com/kubeflow/pipelines/blob/master/samples/contrib/aws-samples/simple_train_pipeline/training-pipeline.py#L76) (uncomment this line)
+
+## Inputs to the pipeline
+
+### Sample Mnist dataset
+
+Use the following python script to copy train_data, test_data, and valid_data to your bucket.  
+[create a bucket](https://docs.aws.amazon.com/AmazonS3/latest/gsg/CreatingABucket.html) in `us-east-1` region if you don't have one already. 
+
+
+Create a new file named s3_sample_data_creator.py with following content :
+```
+import pickle, gzip, numpy, urllib.request, json
+from urllib.parse import urlparse
+
+###################################################################
+# This is the only thing that you need to change to run this code 
+# Give the name of your S3 bucket 
+bucket = '<bucket-name>' 
+
+# If you are gonna use the default values of the pipeline then 
+# give a bucket name which is in us-east-1 region 
+###################################################################
+
+
+# Load the dataset
+urllib.request.urlretrieve("http://deeplearning.net/data/mnist/mnist.pkl.gz", "mnist.pkl.gz")
+with gzip.open('mnist.pkl.gz', 'rb') as f:
+    train_set, valid_set, test_set = pickle.load(f, encoding='latin1')
+
+
+# Upload dataset to S3
+from sagemaker.amazon.common import write_numpy_to_dense_tensor
+import io
+import boto3
+
+train_data_key = 'mnist_kmeans_example/train_data'
+test_data_key = 'mnist_kmeans_example/test_data'
+train_data_location = 's3://{}/{}'.format(bucket, train_data_key)
+test_data_location = 's3://{}/{}'.format(bucket, test_data_key)
+print('training data will be uploaded to: {}'.format(train_data_location))
+print('training data will be uploaded to: {}'.format(test_data_location))
+
+# Convert the training data into the format required by the SageMaker KMeans algorithm
+buf = io.BytesIO()
+write_numpy_to_dense_tensor(buf, train_set[0], train_set[1])
+buf.seek(0)
+
+boto3.resource('s3').Bucket(bucket).Object(train_data_key).upload_fileobj(buf)
+
+# Convert the test data into the format required by the SageMaker KMeans algorithm
+write_numpy_to_dense_tensor(buf, test_set[0], test_set[1])
+buf.seek(0)
+
+boto3.resource('s3').Bucket(bucket).Object(test_data_key).upload_fileobj(buf)
+
+# Convert the valid data into the format required by the SageMaker KMeans algorithm
+numpy.savetxt('valid-data.csv', valid_set[0], delimiter=',', fmt='%g')
+s3_client = boto3.client('s3')
+input_key = "{}/valid_data.csv".format("mnist_kmeans_example/input")
+s3_client.upload_file('valid-data.csv', bucket, input_key)
+```
+Run this file `python s3_sample_data_creator.py`
+
+### Role Input
+
+Run these commands to create the sagemaker-execution-role.   
+Note down the Role ARN. You need to give this Role ARN as input in pipeline.
```

Comment:
Add a line on describing this role: 

This role is needed for sagemaker job to access other AWS resources such as S3 databucket, ECR etc.